components/application-gateway/pkg/authorization/model.go

```diff
@@ -0,0 +1,51 @@
+package authorization
+
+// Credentials contains OAuth or BasicAuth configuration.
+type Credentials struct {
+	// OAuth is OAuth configuration.
+	OAuth *OAuth
+	// BasicAuth is BasicAuth configuration.
+	BasicAuth *BasicAuth
+	// CertificateGen is CertificateGen configuration.
+	CertificateGen *CertificateGen
+	// CSRFTokenEndpointURL (optional) to fetch CSRF token
+	CSRFTokenEndpointURL string
+	// Headers that are injected by the gateway
+	Headers *map[string][]string
+	// QueryParameters that are injected by the gateway
+	QueryParameters *map[string][]string
```

Comment:
As discussed `Headers` and `QueryParameters` are leftovers and we should remove those.

Reply:
Done